allennlp/nn/beam_search.py

```diff
@@ -792,6 +984,13 @@ def _search(
             # shape: (batch_size * beam_size, num_classes)
             class_log_probabilities, state = step(last_predictions, state, timestep + 1)
 
+            for constraint, constraint_state in zip(self.constraints, constraint_states):
+                # shape: (batch_size, beam_size, num_classes)
+                reshaped_class_log_probabilities = class_log_probabilities.view(
+                    batch_size, self.beam_size, -1
+                )
```

Comment:
Any reason not to put the reshape outside of the loop?

```suggestion
            # shape: (batch_size, beam_size, num_classes)
            reshaped_class_log_probabilities = class_log_probabilities.view(
                batch_size, self.beam_size, -1
            )
            for constraint, constraint_state in zip(self.constraints, constraint_states):
```